### ISSUE TITLE
`type(Any)` fails to be assigned to type of `type[Any]` or `type[type]`

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#8973]
**Describe the bug**

With Pyright 1.1.380, `type(Any)` is not assignable to type of `type[Any]` or `type[type]`.

`type(Any)` is currently `typing._AnyMeta` which is a subclass of `type`. Even if the implementation of `Any` got changed, `type(Any)` should still be some subclass of `type`.

**Code or Screenshots**

```python
from typing import Any

AnyMeta = type(Any)
t0: type[Any] = AnyMeta  # Passes with 1.1.379, but fails with 1.1.380.
t1: type[type] = AnyMeta  # Passes with 1.1.379, but fails with 1.1.380.
```

**VS Code extension or command-line**

Pyright CLI 1.1.380

**Comment by erictraut** [comment]
I agree that `AnyMeta` should be assignable to `type[Any]`. I'll fix that.

I don't think that `AnyMeta` should be assignable to `type[type]` for the same reason that `Any` is not assignable to `type`. `Any` is a special form. Its implementation details are not documented or defined in typeshed. Type checkers should therefore not make assumptions about its runtime implementation.

**Comment by Azureblade3808** [comment]
What do you think the type of `type(Any)` is? I guess it should be `type[Any]` or `type[Unknown]`, both of which can be assigned to type `type[type]`. So why is `type(Any)` not assignable to type `type[type]`?

```python
from typing import Any

def foo(a: type, b: type[Any], c: type[type]):
    c = a  # OK
    c = b  # OK
```

**Comment by erictraut** [comment]
`Any` is a special form, and it could be implemented in a variety of ways. The runtime implementation of `Any` isn't documented. The most a static type checker can say about `type(Any)` is that its type is `type[object]`. This type isn't assignable to `type[type]`.

**Comment by Azureblade3808** [comment]
Oh I see. Thanks for explanation.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.381.
